src/EventListener/InsertTagsListener.php

```diff
@@ -155,6 +163,16 @@ private function replaceCompanyInsertTags($insertTag, $field)
 				return $strCountry;
 			case 'countrycode':
 				return Company::get('country');
+            case 'vcard_url':
+                $pageId = 0;
+                $request = $this->requestStack->getCurrentRequest();
+                if (null !== $request) {
+                    /** @var PageModel $page */
+                    $page = $request->attributes->get('pageModel');
+                    $pageId = $page->id;
+                }
+
+                return $this->router->generate('contao_company_vcard_download', ['redirect' => $pageId]);
```

Comment:
Same as in vCardController:
- parameter should be changed to e.g. 'page'. It does not match right now, hence throwing an error